pkg/loop/internal/relayer.go

```diff
@@ -349,11 +354,14 @@ func (r *relayerServer) NewPluginProvider(ctx context.Context, request *pb.NewPl
 func (r *relayerServer) newMedianProvider(ctx context.Context, relayArgs types.RelayArgs, pluginArgs types.PluginArgs) (uint32, error) {
 	i, ok := r.impl.(MedianProvider)
 	if !ok {
-		return 0, errors.New("median not supported")
+		return 0, status.Error(codes.Unimplemented, "median not supported")
```

Comment:
Shouldn't we be able to just wrap the normal error here?
```suggestion
		return 0, fmt.Errorf("median not supported", errors.ErrUnsupported)
```

Reply:
We could, but in this case wouldn't `codes.Unimplemented` be a more specific indication of what the problem is?  The relay hasn't implemented `NewMedianProvider()` so it can't call that method.

Reply:
Also worth noting:  the `errors.ErrUnsupported` type information will be immediately lost as soon as it's sent over the wire, whereas `codes.Unimplemented` will remain and can be detected by the client.  The only way to check for `errors.ErrUnsupported` is with string matching.

Reply:
My thoughts:  I think wrapping it in `errors.ErrUnsupported` isn't helpful... we can either set the code to `Unimplemented` or leave it set to `Unknown`, but either way "median not supported" is about the best we can do for a descriptive error since the error type gets erased anyway.